### PR TITLE
Add importer which imports files from .md/.txt

### PR DESCRIPTION
**Pull request by asmsuechan**
Hi there. I implemented a function which imports files from .md or .txt.

screen shot 2017-03-16 at 13 12 38

### Specifications
* multiple files
* .md or .txt

If someone wants what else, please comment.

### TODO
* fill a title after imported immediately

I'll implement `fill a title` after https://github.com/BoostIO/Boostnote/pull/324 is merged.

### Future
* enable to import files from a directory

### CODE REVIEW

**Comment by sota1235** [comment]
Sorry, firstly plz fix conflicts :bow:

**Comment by asmsuechan** [comment]
Alright. Done.

**Comment by GaryFurash** [comment]
How do I get this? It's not in the current download.

**Comment by sota1235** [comment]
@GaryFurash This feature is in progress now. So wait a moment please... 🙇

**Comment by sota1235** [comment]
@asmsuechan Thanks. I'll check 💡

File: browser/main/NoteList/index.js [hunk 1]
```diff
@@ -349,6 +354,51 @@ class NoteList extends React.Component {
...
+
+    const options = {
+      filters: [
+        { name: 'Documents', extensions: ['md', 'txt']}
+      ],
+      properties: ['openFile', 'multiSelections']
+    }
+
+    const targetIndex = _.findIndex(this.notes, (note) => {
+      return note != null && note.storage + '-' + note.key === location.query.key
```
Comment (sota1235) [inline browser/main/NoteList/index.js]: Can't you use strict equal? `!==` 
And you can use template literal.

File: browser/main/NoteList/index.js [hunk 2]
```diff
@@ -349,6 +354,51 @@ class NoteList extends React.Component {
...
+    })
+
+    const storageKey = this.notes[targetIndex].storage
+    const folderKey = this.notes[targetIndex].folder
+
+    dialog.showOpenDialog(remote.getCurrentWindow(), options, (filepaths) => {
+      if (filepaths === undefined) return
+      filepaths.forEach((filepath) => {
+        fs.readFile(filepath, (err, data) => {
+          if (err) throw Error('File reading error: ', err)
```
Comment (sota1235) [inline browser/main/NoteList/index.js]: [Question] Do you test this situation? Doesn't it crush?

**Comment by jasondavis** [comment]
Just an idea that is related to this....what about the ability to upload a ZIP archive file that contains multiple files of any of these formats,  

- MD
- TXT
- JS
- PHP
- CSS
- HTML
- LESS
- SCSS
- and other code files

Boostnote would then create a Code Snippet File in Boostnote and add a sub-file/tab file for each file in the ZIP archive that is in the allowed formats.

Just an idea

File: browser/main/NoteList/index.js (hunk 1 above)
Comment (asmsuechan) [inline browser/main/NoteList/index.js]: Sure thing.

File: browser/main/NoteList/index.js (hunk 2 above)
Comment (asmsuechan) [inline browser/main/NoteList/index.js]: Umm, I tried by hand, the probability is quite low.

File: browser/main/NoteList/index.js [hunk 3]
```diff
@@ -349,6 +354,51 @@ class NoteList extends React.Component {
...
+          // TODO: fill the title
```
Comment (asmsuechan) [inline browser/main/NoteList/index.js]: I'm gonna open another PR to implement a module for `findTitle()`.

Comment (asmsuechan) [inline browser/main/NoteList/index.js]: I should use the method, so I have to wait for merging it.
https://github.com/BoostIO/Boostnote/pull/548

**Comment by GaryFurash** [comment]
Is this working yet? It doesn't work on my Windows 7 or Windows 10 PC.

**Comment by asmsuechan** [comment]
@GaryFurash  I'm going to try fixing it in a few days.

**Comment by asmsuechan** [comment]
@GaryFurash Well, it seems to work fine for my environment. Please open a new issue with screenshots.

**Pull request merged**